uc-crux-llvm/src/UCCrux/LLVM/Overrides.hs

```diff
@@ -0,0 +1,202 @@
+import           Lang.Crucible.LLVM.TypeContext (TypeContext)
+import           Lang.Crucible.LLVM.Intrinsics (OverrideTemplate(..), register_llvm_overrides, basic_llvm_override)
+
+import           Crux.Types (OverM, Model, HasModel)
+
+-- crux-llvm
+import           Crux.LLVM.Overrides (ArchOk)
+
+-- uc-crux-llvm
+import           UCCrux.LLVM.Errors.Unimplemented (unimplemented)
+import qualified UCCrux.LLVM.Errors.Unimplemented as Unimplemented
+{- ORMOLU_ENABLE -}
+
+newtype UnsoundOverrideName = UnsoundOverrideName {getUnsoundOverrideName :: Text}
+  deriving (Eq, Ord, Show)
+
+-- | Register some additional overrides that are useful for bugfinding, but not
+-- for verification. They unsoundly under-approximate the environment. This
+-- helps symbolic execution reach more code.
+registerUnsoundOverrides ::
+  (ArchOk arch, IsSymInterface sym, HasLLVMAnn sym) =>
+  proxy arch ->
+  L.Module ->
+  ModuleTranslation arch ->
+  IORef (Set UnsoundOverrideName) ->
+  OverM Model sym LLVM ()
+registerUnsoundOverrides proxy llvmModule mtrans usedRef =
+  do
+    let llvmCtx = mtrans ^. transContext
+    let ?lc = llvmCtx ^. llvmTypeCtx
+    register_llvm_overrides llvmModule [] (unsoundOverrides proxy usedRef) llvmCtx
+
+------------------------------------------------------------------------
+
+-- ** Declarations
+
+unsoundOverrides ::
+  ( IsSymInterface sym,
+    HasLLVMAnn sym,
+    ArchOk arch,
+    ?lc :: TypeContext,
+    HasModel personality
+  ) =>
+  proxy arch ->
+  IORef (Set UnsoundOverrideName) ->
+  [OverrideTemplate (personality sym) sym arch rtp l a]
+unsoundOverrides arch usedRef =
+  [ basic_llvm_override $
+      [llvmOvr| i32 @gethostname( i8* , size_t ) |]
+        ( \memOps sym args ->
+            liftIO (used "gethostname")
+              >> Ctx.uncurryAssignment (callGetHostName arch sym memOps) args
+        ),
+    basic_llvm_override $
+      [llvmOvr| i8* @getenv( i8* ) |]
+        ( \memOps sym args ->
+            liftIO (used "getenv")
+              >> Ctx.uncurryAssignment (callGetEnv arch sym memOps) args
+        )
+  ]
+  where
+    used :: Text -> IO ()
+    used name = modifyIORef usedRef (Set.insert (UnsoundOverrideName name))
+
+------------------------------------------------------------------------
+
+-- ** Implementations
+
+callGetHostName ::
+  ( IsSymInterface sym,
+    HasLLVMAnn sym,
+    wptr ~ ArchWidth arch,
+    ArchOk arch,
+    ?lc :: TypeContext
+  ) =>
+  proxy arch ->
+  sym ->
+  GlobalVar Mem ->
+  RegEntry sym (LLVMPointerType wptr) ->
+  RegEntry sym (BVType wptr) ->
+  OverrideSim p sym ext r args ret (RegValue sym (BVType 32))
+callGetHostName _proxy sym mvar (regValue -> ptr) (regValue -> len) =
+  do
+    let hostname = "hostname"
+    let lenLt bv = liftIO (What4.bvSlt sym len =<< What4.bvLit sym ?ptrWidth bv)
+    lenNeg <- lenLt (BV.mkBV ?ptrWidth 0)
+    -- TODO(lb): Panic if ?ptrWidth is like 2 or something crazy
```

Comment:
`?ptrWidth` is at least 16 because `ArchOk` resolves to `HasPtrWidth`

Reply:
Rad, I changed this from a TODO that a NOTE that a check isn't currently necessary. Thanks!